commands/bash_completion.go

```diff
@@ -23,18 +24,20 @@ var bashcompletionCmd = &cobra.Command{
 This currently only works on Bash version 4, and is hidden
 pending a merge of https://github.com/spf13/cobra/pull/520.`,
 	Hidden: true,
-	Run:    runBashcompletion,
+	RunE:   runBashcompletion,
 }
 
-func runBashcompletion(cmd *cobra.Command, args []string) {
+func runBashcompletion(cmd *cobra.Command, args []string) error {
 	if len(args) < 1 {
 		fmt.Println("Please provide filename for bash completion")
```

Comment:
This is a general comment that apply to **ALL** places where an error is returned.

The redundancy in error messaging here is not necessary with Cobra, you should *just* return the error message, and *not* print it.

By using the `RunE` field you are passing the error back to the point where the subcommand was invoked (see the comment on `faas.go`. You should be printing the error message at that point instead of in the subcommand.

Reply:
That is a much cleaner way of doing it than the way I've been doing it. `RunE` does actually print the error messages it receives to screen in a `Error: <Error message here>` format by default unless we suppress it. 

Should we print the messages ourselves or let cobra do it?